### PR TITLE
Lnd-atm coupling fixes

### PR DESCRIPTION
**Pull request by billsacks**
Added atmosphere model's surface height as a required input, and 10-m
wind speed as an output.

Also fixed some typos.

### CODE REVIEW

File: doc/source/tech_note/Ecosystem/CLM50_Tech_Note_Ecosystem.rst [hunk 1]
```diff
@@ -425,6 +425,8 @@ conditions from the current time step.
  +======================================================+================================================+=================================================+
  | :sup:`1`\ Reference height                           | :math:`z'_{atm}`                               | m                                               |
  +------------------------------------------------------+------------------------------------------------+-------------------------------------------------+
+ | Atmosphere model's surface height                    | :math:`z_{surf,atm}`                           | m                                               |
```
Comment (billsacks) [inline doc/source/tech_note/Ecosystem/CLM50_Tech_Note_Ecosystem.rst]: I struggled to come up with a mathematical symbol for this. Please suggest anything else that feels better.

File: doc/source/tech_note/Ecosystem/CLM50_Tech_Note_Ecosystem.rst [hunk 2]
```diff
@@ -566,6 +568,8 @@ calculated from molar ratio and the atmospheric pressure
  +---------------------------------------+------------------------------------------------+--------------------------------------------------------------+
  | Specific humidity at 2 meter height   | :math:`q_{2m}`                                 | kg kg\ :sup:`-1`                                             |
  +---------------------------------------+------------------------------------------------+--------------------------------------------------------------+
+ | Wind speed at 10 meter height         | :math:`u_{10m}`                                | m s\ :sup:`-1`                                               |
```
Comment (billsacks) [inline doc/source/tech_note/Ecosystem/CLM50_Tech_Note_Ecosystem.rst]: It seems like this was accidentally left out, but I'd like confirmation that this is, indeed, a standard output (it appears in subroutine lnd_export, but I don't know if I'm missing something).

**Comment by billsacks** [comment]
@olyson I'd like your sign-off on these changes, then I'll merge to the main gh-pages branch.

**Comment by billsacks** [comment]
Thanks, @olyson . I'll merge this to the main branch, then let you update the html at your convenience. (Based on the wiki documentation, it sounds like that's the standard procedure, right?)

Right that surface height is only used by the glacier model right now. I still need to update that section of the documentation... I'll try to remember to be consistent :-)

**Comment by olyson** [comment]
Yep, I'll update the html soon, thanks!

**Pull request merged**